### PR TITLE
support managed pools

### PR DESCRIPTION
**Pull request by mendesfabio**
# Description

This PR adds support to ManagedPool V2. I deployed the Subgraph to Ethereum and Goerli on my side and performed some tests to ensure events are correctly handled. Examples:

- [Add and remove tokens](https://api.thegraph.com/subgraphs/name/mendesfabio/balancer-goerli-v2/graphql?query=%7B%0A++creation:+pool(%0A++++id:+%220x03188c97f878257a417e4679ace013df01c62414000100000000000000000778%22%0A++%09block:+%7B%0A++++++number:+8858157%0A++++%7D%0A++)+%7B%0A++++poolType%09%0A++++poolTypeVersion%0A++++tokensList%0A++%7D%0A++addToken:+pool(%0A++++id:+%220x03188c97f878257a417e4679ace013df01c62414000100000000000000000778%22%0A++%09block:+%7B%0A++++++number:+8883929%0A++++%7D%0A++)+%7B%0A++++poolType%09%0A++++poolTypeVersion%0A++++tokensList%0A++++tokens(orderBy:+index)+%7B%0A++++++address%0A++++++index%0A++++%7D%0A++%7D%0A++removeToken:+pool(%0A++++id:+%220x03188c97f878257a417e4679ace013df01c62414000100000000000000000778%22%0A++%09block:+%7B%0A++++++number:+8935431%0A++++%7D%0A++)+%7B%0A++++poolType%09%0A++++poolTypeVersion%0A++++tokensList%0A++++tokens(orderBy:+index)+%7B%0A++++++address%0A++++++index%0A++++%7D%0A++%7D%0A++addToken2:+pool(%0A++++id:+%220x03188c97f878257a417e4679ace013df01c62414000100000000000000000778%22%0A++%09block:+%7B%0A++%09number:+8935436%0A++%09%7D%0A++)+%7B%0A++++poolType%09%0A++++poolTypeVersion%0A++++tokensList%0A++++tokens(orderBy:+index)+%7B%0A++++++address%0A++++++index%0A++++%7D%0A++%7D%0A%7D%0A)
- [Fees collected in BPT](https://api.thegraph.com/subgraphs/name/mendesfabio/balancer-goerli-v2/graphql?query=%7B%0A++creation:+pool(%0A++++id:+%220x03188c97f878257a417e4679ace013df01c62414000100000000000000000778%22%0A++%09block:+%7B%0A++++++number:+8858157%0A++++%7D%0A++)+%7B%0A++++poolType%09%0A++++poolTypeVersion%0A++++totalAumFeeCollectedInBPT%0A++%7D%0A++feesCollected:+pool(%0A++++++id:+%220x03188c97f878257a417e4679ace013df01c62414000100000000000000000778%22%0A++++)+%7B%0A++++%09totalAumFeeCollectedInBPT%0A++%7D%0A%7D%0A)
- [Protocol fee cache](https://api.thegraph.com/subgraphs/name/mendesfabio/balancer-v2/graphql?query=%7B%0A%09pools+%7B%0A%09++id%0A++++protocolAumFeeCache%0A++++protocolSwapFeeCache%0A++++protocolYieldFeeCache%0A%09%7D%0A%7D%0A)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by rehan509** [comment]
@mendesfabio this is not working as I deployed this on Arbitrium Subgraph , It failed on 83% sync

**Comment by rehan509** [comment]
> > @mendesfabio this is not working as I deployed this on Arbitrium Subgraph , It failed on 83% sync
> 
> hey @rehan509 could you send us the link to the deployment?

@mendesfabio  https://thegraph.com/hosted-service/subgraph/muhammadwajidshahid/profilemanager    fail link         handleEncodedProtocolFeePercentageCacheUpdated this function was failing.
The problem was in the toI32.                        Now it is working https://thegraph.com/hosted-service/subgraph/rehan509/my-fair

File: schema.graphql [hunk 1]
```diff
@@ -144,6 +151,12 @@ type PoolToken @entity {
 
   # ComposableStablePool Only
   isExemptFromYieldProtocolFee: Boolean
+
+  # ManagedPool Only
+  circuitBreaker: CircuitBreaker
+
+  # FXPool Only
+  assimilator: Bytes
```
Comment (markusbkoch) [inline schema.graphql:159]: This looks like it might have been inadvertently introduced in this PR?

File: src/mappings/poolController.ts [hunk 2]
```diff
@@ -67,6 +194,37 @@ export function handleProtocolFeePercentageCacheUpdated(event: ProtocolFeePercen
...
+  let pool = Pool.load(poolContract.pool) as Pool;
+
+  // Convert to hex and remove the 0x prefix
+  const feeCache = event.params.feeCache.toHex().slice(2);
+
+  // Each byte represents 2 hex digits
+  // Thus each fee is represented by 16 chars
+  let encodedAumFee = feeCache.slice(16, 32);
+  let encodedYieldFee = feeCache.slice(32, 48);
+  let encodedSwapFee = feeCache.slice(48, 65);
```
Comment (markusbkoch) [inline src/mappings/poolController.ts:215]: Why is this 17 and not 16 like the others? IIUC it doesn't matter for `slice` if `end>len`, so it works anyway? But it looks strange.

**Review by markusbkoch — APPROVED** [review]
A couple of minor comments, looks good

File: src/mappings/poolController.ts (hunk 2 above)
Comment (mendesfabio) [reply src/mappings/poolController.ts]: yea, I shouldn't have committed this. I fixed and re-deployed and still works as expected.

**Comment by rehan509** [comment]
ok @mendesfabio  thanks  but i had resolved the issue using Subsquid  .

**Pull request merged**